lib/valueflow.cpp

```diff
@@ -2920,6 +2920,18 @@ static bool valueFlowForwardVariable(Token* const startToken,
                 return false;
         }
 
+        // Function call
+        else if (!var->isLocal() && tok2->function() && Token::Match(tok2, "%name% (")) {
+            if (tok2->function()->functionScope) {
+                const bool c = isVariableChanged(
```

Comment:
We should only check if the variable is changed if the member function call is non-const. It should be something like: `!tok->function()->isConst() || isVariableChanged(...)`.

Reply:
Done